scripts/spack_manifest/injection/modules.py

```diff
@@ -0,0 +1,279 @@
+import argparse
+import yaml
+import re
+import sys
+
+from typing import Any
+
+##################
+# Main functions #
+##################
+
+def main():
+    # Get inputs
+    args = parse_args(sys.argv[1:])
+
+    packages: set[str] = set(args.packages.split())
+
+    with open(args.manifest, "r") as file:
+        manifest: dict[str, Any] = yaml.safe_load(file)
+
+    # Inject manifeest with projections and includes
+    manifest_with_projections: dict[str, Any] = inject_projections(
+        manifest=manifest, root_spec=args.root_spec, packages=packages
+    )
+
+    manifest_with_projections_and_includes: dict[str, Any] = inject_includes(manifest=manifest_with_projections, root_spec=args.root_spec, packages=packages)
+
+    # Output the modified manifest
+    print(
+        yaml.dump(
+            manifest_with_projections_and_includes,
+            default_flow_style=False,
+            sort_keys=False,
+        )
+    )
+
+    if args.output:
+        with open(args.output, "w") as output_file:
+            yaml.dump(
+                manifest_with_projections_and_includes,
+                output_file,
+                default_flow_style=False,
+                sort_keys=False,
+            )
+
+def inject_projections(
+    manifest: str, root_spec: str, packages: set[str]
+) -> dict[str, Any]:
+
+    # Get projections that are already defined in the manifest - we don't want to redefine these
+    defined_projections: dict[str, str] = _get_defined_projections(manifest)
+    defined_projections_set: set[str] = set(defined_projections.keys())
+    # Get packages that have versions defined in the manifest - we can only generate projections for things with an explicit version
+    packages_with_versions_defined: set[str] = _get_packages_with_versions_defined(
+        manifest
+    )
+
+    # Essentially...we want to generate projections for all packages that don't already have them, provided they have a version defined
```

Comment:
```suggestion
    # Generate projections for all packages that don't already have them, provided they have a version defined
```
I'm a fan of relaxed conversational chat, but in this case I think it obscures meaning, and we should always have in mind those for whom English is not their primary language and keep it as clear as possible.